### PR TITLE
Allow change value from parent

### PR DESCRIPTION
**Pull request by gdotdesign**
This is basically #9. 

For me it works perfectly but I don't know if it covers every use case.

Also added two small features:
- Empty string can be supplied as value (to be able to show placeholder)
- Delegate placeholder attribute from the props

### CODE REVIEW

**Comment by uhlryk** [comment]
thanks It looks super.

**Comment by uhlryk** [comment]
I will add to documentation (README.md) info about placeholder prop. And increase version to 1.3.0 (version change is needed to publish to npm).

**Comment by gdotdesign** [comment]
Thanks :+1:

**Comment by uhlryk** [comment]
And I will add more tests.

**Comment by uhlryk** [comment]
I made small changes in your code
1. [link](https://github.com/uhlryk/react-dynamic-number/blob/master/src/index.jsx#L54) changed `==` to `===` more strict
2. [link](https://github.com/uhlryk/react-dynamic-number/blob/master/src/index.jsx#L59) because parent render can trigger this component `componentWillReceiveProps` method. And if component has not value prop then it was reset to 0 - it works like controlled component without control :) After changes only components with prop value are controlled components (as normal react inputs).
How works controlled components https://facebook.github.io/react/docs/forms.html#controlled-components

We can add defalutValue prop too. It will set value once

**Comment by uhlryk** [comment]
I'm not sure about this line : [link](https://github.com/uhlryk/react-dynamic-number/blob/master/src/index.jsx#L60)
I didn't tested it yet, maybe I m wrong:
`this.calculator.calculate` is designed to test raw string values if they have certain format and to change them to js numbers. For example If we set in props decimal separator comma, thousand separator dot. And we pass normal javascript float 1234.1234 to `calculate` it will treat this value as raw string and remove dot (because dot in this configuration means thousand separator).
If Im right then we need new method similar to `calculate` which transform correct js numbers to certain string view values.

**Comment by gdotdesign** [comment]
- Yeah adding `defaultValue` makes sense.
- Maybe it's enough to use the `_createViewValue` function because we don't need to decode the `nextProps.value` because it's already a number?

**Comment by uhlryk** [comment]
> Yeah adding defaultValue makes sense.

I have added ticket

> Maybe it's enough to use the _createViewValue function because we don't need to decode the nextProps.value because it's already a number?

this is good start. but `_createViewValue` didn't change decimal separator, because `calculate` method expect raw value from input with selected in props decimal separator character (dot or comma). And if component has comma as decimal separator and we pass correct js number (dot separator).
We have to create method which have opposite functionality to `_createModelValue` and then invoke `_createViewValue`. 

In other words this component have to work with three types of numbers:
1. correct js number (model value)
2. number which is written in input field (raw value)
3. number which is defined by props (view value)
In calculation, function tries to remove wrong characters and recalculate raw number to model and view values. But it also do regexp tests and if user write wrong characters like letters it will rollback view and model to previous state.
